pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -490,14 +493,22 @@ func (r *Reconciler) getOrCreateClusterResourceSnapshot(ctx context.Context, crp
 	// mark the last resource snapshot as inactive if it is different from what we have now or 3) when some
 	// sub-indexed cluster resource snapshots belonging to the same group have not been created, the master
 	// cluster resource snapshot should exist and be latest.
-	if latestResourceSnapshot != nil &&
-		latestResourceSnapshotHash != resourceHash &&
-		latestResourceSnapshot.Labels[fleetv1beta1.IsLatestSnapshotLabel] == strconv.FormatBool(true) {
+	if latestResourceSnapshot != nil && latestResourceSnapshotHash != resourceHash && latestResourceSnapshot.Labels[fleetv1beta1.IsLatestSnapshotLabel] == strconv.FormatBool(true) {
```

Comment:
Can we also add one integration test? Since our e2e test never tests the requeue behavior, I think it would be good to have: create a CRP -> verify one resource snapshot generated -> add/update some resources -> verify only one new resource snapshot generated and the difference between the two creationTimestamp is larger than the interval.